core/setMetExchange.m

```diff
@@ -0,0 +1,212 @@
+function [exchModel,unusedMets] = setMetExchange(model,mets,lb,ub,closeOthers,mediaOnly)
+% setMetExchange
+%   Define the exchange flux bounds for a set of metabolites.
+%
+% Input:
+%   model         a model structure
+%   mets          a cell array of metabolite names (case insensitive) or 
+%                 metabolite IDs, or a vector of metabolite indices
+%                 (opt, default all exchanged metabolites)
+%   lb            lower bound of exchange flux. Can be either a vector of
+%                 bounds corresponding to each of the provided metabolites,
+%                 or a single value that will be applied to all.
+%                 (opt, default -1000)
+%   ub            upper bound of exchange flux. Can be either a vector of
+%                 bounds corresponding to each of the provided metabolites,
+%                 or a single value that will be applied to all.
+%                 (opt, default 1000)
+%   closeOthers   close exchange reactions for all other exchanged 
+%                 metabolites not present in the provided list. This will
+%                 prevent IMPORT of the metabolites, but their EXPORT will
+%                 not be modified.
+%                 (opt, default true)
+%   mediaOnly     only consider exchange reactions involving exchange to or
+%                 from the extracellular (media) compartment. Reactions
+%                 such as "sink" reactions that exchange metabolites
+%                 directly with an intracellular compartment will therefore
+%                 be ignored even though "getExchangeRxns" identifies such
+%                 such reactions as exchange reactions.
+%                 Note: The function will attempt to identify the
+%                 extracellular compartment by the "compNames" field, and
+%                 also requires the "metComps" field to be present,
+%                 otherwise the mediaOnly flag will be ignored.
+%                 (opt, default false)
+%
+% Output:
+%   exchModel     a model structure with updated exchange flux bounds for
+%                 the provided set of metabolites
+%   unusedMets    metabolites provided by the user that were not used
+%                 because they are not involved in any exchange reactions
+%                 in the model
+%
+% NOTE: Exchange reactions involving more than one metabolite will be
+% ignored.
+%
+% Usage: exchModel = setMetExchange(model,mets,lb,ub,closeOthers,mediaOnly);
+%
+%
+% Jonathan Robinson, 2019-05-20
+%
+
+
+% handle input arguments
+if nargin < 2
+    mets = [];
+elseif ischar(mets)
+    mets = {mets};  % in case only one metabolite is provided as a string
+end 
+if nargin < 3 || isempty(lb)
+    lb = -1000;
```

Comment:
Instead of using -1000 and 1000 as default LB and UB, models can have different default values set under `model.annotation.defaultLB` and `model.annotation.defaultUB`. So if now lb or ub is defined when running this function, first check if default bounds are defined for the model, while only as last resort use -1000 and 1000 as default values.

Reply:
Oh, nice, I didn't know that was a possible field. I've now modified the function to check for this field before defaulting to +/-1000.